Add a fallback condition for musl ldd check

am11 opened the pull request:
If `ldd` fails with `Exec format error`, attempt to locate `musl` token
in the output of `strings /path/ldd`.

am11 commented on eng/native/init-distro-rid.sh:
```diff
@@ -138,7 +138,8 @@ initDistroRidGlobal()
         # Check for musl-based distros (e.g Alpine Linux, Void Linux).
-        if "${rootfsDir}/usr/bin/ldd" --version 2>&1 | grep -q musl; then
+        if "${rootfsDir}/usr/bin/ldd" --version 2>&1 | grep -q musl ||
+           file "${rootfsDir}/usr/bin/ldd" 2>&1 | grep -q musl; then
```
Could also check the same thing with `strings`:

```sh
if "/crossrootfs/arm64/usr/bin/ldd" --version 2>&1 | grep -q musl ||
    strings "/crossrootfs/arm64/usr/bin/ldd" 2>&1 | grep -q musl ||
    file "/crossrootfs/arm64/usr/bin/ldd" 2>&1 | grep -q musl; then
```

janvorli replied:
I would keep just the way with "strings (without even trying to run ldd). It should cover all the cases

am11 replied:
@janvorli, unfortunately in voidlinux `strings(1)` and `file(1)` are not available in the base image. So keeping the --version checking would help. `file` is redundant, so I have removed it.

janvorli replied:
Ok, makes sense.

dagood commented on eng/native/init-distro-rid.sh:
```diff
@@ -138,7 +138,8 @@ initDistroRidGlobal()
         # Check for musl-based distros (e.g Alpine Linux, Void Linux).
-        if "${rootfsDir}/usr/bin/ldd" --version 2>&1 | grep -q musl; then
+        if "${rootfsDir}/usr/bin/ldd" --version 2>&1 | grep -q musl ||
+           strings "${rootfsDir}/usr/bin/ldd" 2>&1 | grep -q musl; then
```
(Nit) This looks like a missing space mistake on first glance... but I suppose it's column alignment?

I don't really have any specific suggestion--I don't know if this is considered readable in same-line-brace styles like we're using here. 😕 I have seen double-indent used on the continuation line to clarify this situation.

am11 replied:
Yup, it was to align it with start of condition; changed to double aligned (4+4 spaces) with respect to the `if` token.

am11 committed:
Add a fallback condition for musl ldd check
If ldd fails with Exec format error, attempt to locate musl token
in the output of `strings /path/ldd`.

safern commented:
Failures are unrelated and known. Merging.
